service/src/test/java/uk/nhs/adaptors/gp2gp/common/storageconnectors/StorageConnectorFactoryTest.java

```diff
@@ -0,0 +1,44 @@
+package uk.nhs.adaptors.gp2gp.common.storageconnectors;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.util.ReflectionTestUtils;
+import static org.mockito.Mockito.when;
+import static org.assertj.core.api.Assertions.assertThat;
+import java.util.stream.Stream;
+
+@ContextConfiguration
+@SpringBootTest
+public class StorageConnectorFactoryTest {
+
+    @InjectMocks
+    private StorageConnectorFactory storageConnectorFactory;
+    @Mock
+    private StorageConnectorConfiguration configuration;
+
+    @ParameterizedTest
+    @MethodSource("provideConnectorTypesForTest")
+    public void When_PlatformIsLocal_Expect_LocalMockConnector(String platform, Class connectorClass) throws Exception {
```

Comment:
After consulting this with @Broom94 we decided that unit test would not correctly test the logic behind the factory operation. A component test would be more fitting in this case.